src/Storages/ObjectStorage/DataLakes/DataLakeConfiguration.h

```diff
@@ -329,10 +347,52 @@ class DataLakeConfiguration : public BaseStorageConfiguration, public std::enabl
         current_metadata->addDeleteTransformers(object_info, builder, format_settings, local_context);
     }
 
+    void fromDisk(ASTs & args, ContextPtr context, bool with_structure) override
+    {
+        auto disk = context->getDisk(context->getSettingsRef()[Setting::iceberg_disk_name].value);
+        ready_object_storage = disk->getObjectStorage();
+
+        if (args.size() > BaseStorageConfiguration::getMaxNumberOfArguments(with_structure))
+            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                "Storage Local requires 1 to {} arguments. All supported signatures:\n{}",
+                BaseStorageConfiguration::getMaxNumberOfArguments(with_structure),
```

Comment:
I guess we should not allow user to specify auth specific args, so using `getMaxNumbferOfArguments` from base configuration does not look correct.